src/DataViews/PresetsDataView.cpp

```diff
@@ -196,9 +206,22 @@ void PresetsDataView::OnShowInExplorerRequested(const std::vector<int>& selectio
 }
 
 void PresetsDataView::OnDoubleClicked(int index) {
-  const PresetFile& preset = GetPreset(index);
+  PresetFile& preset = GetMutablePreset(index);
   if (app_->GetPresetLoadState(preset).state != PresetLoadState::kNotLoadable) {
-    app_->LoadPreset(preset);
+    app_->LoadPreset(preset).Then(main_thread_executor_.get(),
+                                  [&preset](ErrorMessageOr<void> result) {
+                                    if (!result.has_error()) preset.SetIsLoaded(true);
```

Comment:
If you want to do something just in the not error case, you can use `ThenIfSuccess` instead of `Then`